diff --git a/msi-ec.c b/msi-ec.c
index 406f493..98b14a9 100644
--- a/msi-ec.c
+++ b/msi-ec.c
@@ -45,6 +45,7 @@
 #include <linux/seq_file.h>
 #include <linux/string.h>
 #include <linux/slab.h>
+#include <linux/version.h>
 
 static const char *const SM_ECO_NAME       = "eco";
 static const char *const SM_COMFORT_NAME   = "comfort";
@@ -923,14 +924,22 @@ static struct attribute *msi_battery_attrs[] = {
 
 ATTRIBUTE_GROUPS(msi_battery);
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6,2,0)
 static int msi_battery_add(struct power_supply *battery,
 			   struct acpi_battery_hook *hook)
+#else
+static int msi_battery_add(struct power_supply *battery)
+#endif
 {
 	return device_add_groups(&battery->dev, msi_battery_groups);
 }
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6,2,0)
 static int msi_battery_remove(struct power_supply *battery,
 			      struct acpi_battery_hook *hook)
+#else
+static int msi_battery_remove(struct power_supply *battery)
+#endif
 {
 	device_remove_groups(&battery->dev, msi_battery_groups);
 	return 0;
